test/e2e/docs/patterns_tricks_gotchas.md

```diff
@@ -0,0 +1,89 @@
+<div style="width: 45%; float:left" align="left"><a href="./style_guide.md"><-- Style Guide</a> </div>
+<div style="width: 5%; float:left" align="center"><a href="./../README.md">Top</a></div>
+<div style="width: 45%; float:right"align="right"><a href="./debugging.md">Debugging --></a> </div>
+
+# Patterns, Tricks, and Gotchas
+
+<!-- TOC -->
+
+- [Patterns, Tricks, and Gotchas](#patterns-tricks-and-gotchas)
+  - [Asserting against page content](#asserting-against-page-content)
+    - [Returned values and Jest expect()](#returned-values-and-jest-expect)
+	- [Using waitForSelector, text selectors, and 'validate*' methods](#using-waitforselector-text-selectors-and-validate-methods)
+  - [Race conditions with handler registration](#race-conditions-with-handler-registration)
+  - [Handling Calypso navigation](#handling-calypso-navigation)
+
+<!-- /TOC -->
+
+## Asserting against page content
+
+Let's say you want to validate some state or content on a page in Calypso. How should you go about doing that? There are two patterns that are used throughout our E2E tests.
+
+### Returned values and Jest expect()
+
+If it's easy to find and return some discrete piece of information from the page under test, you can return that discrete value from the POM class function, and assert against it using the Jest `expect` library.
+
+Example:
+
+```typescript
+// In the POM class...
+async getActiveTabName(): Promise< string > {
+	return 'foo';
+}
+
+// In the test spec...
+it( 'Active tab is "foo"', async function () {
+	expect( await pomClass.getActiveTabName() ).toBe( 'foo' );
+} );
+```
+
+**Be Warned!** This is not always as easy as it seems! Many of Playwright's functions that check state (like `isVisible()`) return _immediately_. This can be really tricky with async React apps like Calypso, which often require some measure of waiting for state to be propogated and components to be rendered.
```

Comment:
I'm surprised the `**Be Warned**` part didn't fail linting, something about not using bold to do headings.

Reply:
Ah sure, I removed the bolding! 👍